app/pages/lab/actions/media.js

```diff
@@ -5,18 +5,33 @@ const putFile = require('../../../lib/put-file');
 const MAX_FILE_SIZE = 500 * 1024;
 
 const mediaActions = {
-  fetchMedia(props = this.props) {
-    this.setState({ media: null });
-
-    return props.resource.get(this.props.link, { page_size: this.props.pageSize })
+  fetchMedia(props = this.props, page = 1) {
+    this.setState({ media: null, page });
```

Comment:
`setState` should be in the React component. I think the action here should just be an asynchronous function that resolves with a page of media (or empty array on error.)